### PR TITLE
New detected lane model (related to Handling Junctions with OSI::GroundTruth::Lane - issue #80 and pull request #110)

### PR DESCRIPTION
**Pull request by carsten-kuebler**
According to the OSI workshop in Ulm (November 2017), I propose this new detected lane model and the corresponding OSI messages. Change break downward compatibility.  Changes are similar to #110 (new Lane model).

 issue #80, pull request #110

### CODE REVIEW

File: osi_detectedlane.proto [hunk 1]
```diff
@@ -24,4 +24,89 @@ message DetectedLane
...
+//
+message DetectedLaneBoundary
+{
+    // The id of the original laneboundary in the ground truth.
+    // In case of a ghost laneboundary (no corresponding ground truth), this field should be unset.
+    optional Identifier ground_truth_id = 1;
+
+    // The lane boundary to which this detection is associated to.
+    // 
+    optional LaneBoundary laneboundary = 2;
```
Comment (CarloVanDriestenBMW) [inline osi_detectedlane.proto]: lane_boundary ?

File: osi_detectedlane.proto [hunk 2]
```diff
@@ -24,4 +25,47 @@ message DetectedLane
     // Estimated probability that this lane really exists. Range [0,1].
     //
     optional double existence_probability = 3;
+
+    // The measurement state of the detected lane.
+    //
+    optional DetectedTrafficSign.MeasurementState measurement_state = 4;
```
Comment (pmai) [inline osi_detectedlane.proto]: I think if the message is to be reused, it should be separated out from the DetectedTrafficSign message (it probably makes sense to have this centrally defined; I can provide a patch for this if agree upon). @CarloVanDriestenBMW ?

Comment (pmai) [reply osi_detectedlane.proto]: See my comments on PR #117 , I'd propose (and do this during merging) to move MeasurementState into DetectedObjectHeader, and move DetectedObjectHeader out of DetectedObject, which should handle all of this cleanly, I think. @CarloVanDriestenBMW ?

**Comment by pmai** [comment]
Further refactoring is handled in PR #117, so merging for now.

**Pull request merged**